src/tigerbeetle-java/src/main/java/com/tigerbeetle/Transfers.java

```diff
@@ -0,0 +1,460 @@
+     *
+     * @param ledger a 32-bit integer.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setLedger(final int ledger) {
+        putUInt32(at(Struct.Ledger), ledger);
+    }
+
+    /**
+     * Gets a user-defined enum denoting the reason for (or category of) the transfer.
+     *
+     * @return a 16-bit unsigned integer.
+     */
+    public int getCode() {
+        return getUInt16(at(Struct.Code));
+    }
+
+    /**
+     * Sets a user-defined enum denoting the reason for (or category of) the transfer.
+     * <p>
+     * Must be non-zero.
+     *
+     * @param code a 16-bit unsigned integer defined by the user.
+     * @throws IllegalArgumentException if code is negative or greater than 65535.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setCode(final int code) {
+        putUInt16(at(Struct.Code), code);
+    }
+
+    /**
+     * Gets a bitfield that specifies (optional) transfer behavior.
+     *
+     * @see com.tigerbeetle.TransferFlags
+     * @return a 16-bit unsigned integer bit mask.
+     */
+    public int getFlags() {
+        return getUInt16(at(Struct.Flags));
+    }
+
+    /**
+     * Sets a bitfield that specifies (optional) transfer behavior.
+     *
+     * @see com.tigerbeetle.TransferFlags
+     * @param flags a 16-bit unsigned integer bit mask.
+     * @throws IllegalArgumentException if flags is negative or greater than 65535.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setFlags(final int flags) {
+        putUInt16(at(Struct.Flags), flags);
+    }
+
+    /**
+     * Gets how much should be debited from the {@link #setDebitAccountId debit account} and
+     * credited to the {@link #setCreditAccountId credit account}.
+     * <p>
+     * Must always be interpreted as a positive integer.
+     *
+     * @return A 64-bit integer.
+     */
+    public long getAmount() {
+        return getUInt64(at(Struct.Amount));
+    }
+
+    /**
+     * Sets how much should be debited from the {@link #setDebitAccountId debit account} and
+     * credited to the {@link #setCreditAccountId credit account}.
+     * <p>
+     * Must always be interpreted as a positive integer.
+     *
+     * @param amount a 64-bit integer.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setAmount(final long amount) {
+        putUInt64(at(Struct.Amount), amount);
+    }
+
+    /**
+     * Gets the time the transfer was created.
+     * <p>
+     * This is set by TigerBeetle. The format is UNIX timestamp in nanoseconds.
+     *
+     * @return a 64-bit integer.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public long getTimestamp() {
+        return getUInt64(at(Struct.Timestamp));
+    }
+
+    void setTimestamp(final long timestamp) {
```

Comment:
This can only ever be zero from the client. Maybe don't even expose this at all?

Reply:
Yes, you're right, that's not public. From the user's perspective, it's a read-only property.

I just have to leave the `setTimestamp()` method internal because we need it for mocking values for the tests, otherwise, it could be completely removed.